velox/type/Timestamp.h

```diff
@@ -33,9 +33,14 @@ namespace facebook::velox {
 struct Timestamp {
  public:
   enum class Precision : int { kMilliseconds = 3, kNanoseconds = 9 };
+  // nanos in Timestamp should not greater than 1 second.
+  static constexpr uint64_t MAX_NANOS = 999'999'999;
+
   constexpr Timestamp() : seconds_(0), nanos_(0) {}
-  constexpr Timestamp(int64_t seconds, uint64_t nanos)
-      : seconds_(seconds), nanos_(nanos) {}
+  Timestamp(int64_t seconds, uint64_t nanos)
+      : seconds_(seconds), nanos_(nanos) {
+    VELOX_CHECK_LE(nanos, MAX_NANOS);
```

Comment:
I wonder if this check may end up in some hot paths... CC: @Yuhta @laithsakka

Reply:
Yes it might.  Maybe `VELOX_DCHECK`?

Reply:
@liujiayi771 Let's use VELOX_DCHECK (debug-only) here.